extensions/VisualEditor/wikia/ApiAddMediaTemporary.php

```diff
@@ -43,7 +43,7 @@ private function executeImage() {
 			// If wiki is Japanese content, then we check if anonymous edit is allowed. INT-158
 			// This condition will be changed as soon as Mercury has login for all wikis.
 			if ( $wgContLanguageCode === 'ja' ) {
-				if ( $wgDisableAnonymousEditing ) {
+				if ( $wgDisableAnonymousEditing && !$this->mUser->isLoggedIn()) {
```

Comment:
This will work for this particular issue - however I'm not a fond of this solution - let me explain why.
Originally this code was just using checkPermissions() method to figure out if it should proceed forward or not. Later on we decided that we want to make an exception for JA and we did - by just not calling checkPermissions in that particular case. But now it seems like you are not just making an exception (as possibly you could) but instead implementing another path that is checking permissions.
Wouldn't this just work:
```
if ( ! ( $wgContLanguageCode === 'ja' && !$wgDisableAnonymousEditing  ) ) {
    $this->checkPermissions();
}
```